dora/core/client/fs/src/main/java/alluxio/client/file/DoraCacheFileSystem.java

```diff
@@ -139,7 +138,7 @@ public URIStatus getStatus(AlluxioURI path, GetStatusPOptions options)
     } catch (RuntimeException ex) {
       if (ex instanceof StatusRuntimeException) {
         if (((StatusRuntimeException) ex).getStatus().getCode() == Status.NOT_FOUND.getCode()) {
-          throw new FileNotFoundException();
+          throw new FileDoesNotExistException(ufsFullPath);
```

Comment:
@huanghua78 FYI, in the testing, I found the CpCommand expects the FileDoesNotExistException instead of FileNotFoundException.

Reply:
We need to fix all the inconsistent cases in our code.
FileDoesNotExistException is an AlluxioException, but FileNotFoundException is a java.io.FileNotFoundException.
FileNotFoundException is more generic, and some application expects this exception, especially if we return this exception to user applications.

If possible, I'd suggest to return FileNotFoundException in this case, and fix 'CpCommand'.

Reply:
that's right, and actually it's not a single issue just for CpCommand. In the `alluxio.client.file.FileSystem#getStatus(alluxio.AlluxioURI)`, the definition of the method declares it would throw FileNotFoundException currently. That's why CpCommand expects the FileNotFoundException.
Considering the FileNotFoundException is an IOException, we may need to remove the FileNotFoundException from the throws list of the interface to avoid misunderstanding or misuse. What do you think about it?

Reply:
I checked the code in https://github.com/Alluxio/alluxio/blob/main/dora/core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java#L363-L364. I would suggest we keep the FileDoesNotExistException here. If the application use the compatibal filesystem, they will convert the alluxio exception to IOException. If the application use the alluxio client directly, they need to process the alluxio exception.

Reply:
Most likely it is a fundamental exception logic.  
In my impression there is a boundary where to convert the alluxio internal exception to a generic exception, in that box, the alluxio internal exception can be used.
Most likely that is not a small effort, I would suggest adding a dedicate effort to handle that.